README.md

```diff
@@ -215,7 +215,7 @@ Start the daemon.
 
 `flags` - Flags array to be passed to the `ipfs daemon` command.
 
-`callback` is a function with the signature `function(err, ipfsApi)` that receives an instance of `ipfs-api` on success or an instance of `Error` on failure
+`callback` is a function with the signature `function(Error, ipfsApi)` that receives an instance of `Error` on failure or an instance of `ipfs-api` on success. 
```

Comment:
The changes you've made in the other files do not return an instance of `ipfs-api`.

Reply:
@alanshaw these changes are the docs for `start`. His changes in the other files are for the `init`

Reply:
Ah sorry!

Reply:
Yeah included also this change because I initially thought start method was also not returning correctly.

Reply:
I think start should also return ipfsd instead of API.
ipfsd includes the api and waterfalling spawn,init,start for non disposable gets easier. What you guys think?

Reply:
I agree that it should return the `ipfsd`, as it has the `api`. However, this would probably be a breaking change in a lot of repos...

Reply:
yes :/ i'll create an issue about that